compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala

```diff
@@ -51,10 +52,21 @@ class PruneErasedDefs extends MiniPhase with SymTransformer { thisTransform =>
     else cpy.ValDef(tree)(rhs = trivialErasedTree(tree.rhs))
 
   override def transformDefDef(tree: DefDef)(using Context): Tree =
+    checkNoInlineAnnoClasses(tree)
     checkErasedInExperimental(tree.symbol)
     if !tree.symbol.isEffectivelyErased || tree.rhs.isEmpty then tree
     else cpy.DefDef(tree)(rhs = trivialErasedTree(tree.rhs))
 
+  def checkNoInlineAnnoClasses(tree: DefDef)(using Context): Unit =
+    if tree.symbol.is(Inline) then
+      new TreeTraverser {
+        def traverse(tree: Tree)(using Context): Unit =
+          tree match
+            case tree: TypeDef if tree.symbol.isAnonymousClass =>
+              report.warning(new InlinedAnonClassWarning(), tree.symbol.sourcePos)
+            case _ => traverseChildren(tree)
+      }.traverse(tree)
+
```

Comment:
I don't think this is the correct place to put this check in, I think it makes more sense to add it in [`RefChecks`](https://github.com/scala/scala3/blob/05354ba5317ad4052505b04c6aec2fbb07264f89/compiler/src/dotty/tools/dotc/typer/RefChecks.scala)'.

https://github.com/scala/scala3/blob/05354ba5317ad4052505b04c6aec2fbb07264f89/compiler/src/dotty/tools/dotc/Compiler.scala#L72

Reply:
Sure, I don't feel strongly so I have moved it, although if this method is ever called from somewhere other than where it was originally defined then the warning itself would very likely need to change.